superdwarf-lua.lua

```diff
@@ -0,0 +1,126 @@
+-- makes units very speedy
+--@ module = true
+local help = [====[
+
+superdwarf
+==============
+Similar to `fastdwarf`, per-creature.
+Overrides units timers every game tick to 1, causing extremely fast actions
+
+Usage:
+
+Valid Options:
+
+``-unit <id>``: Targets the unit with the specified ID.
+                This is optional; if not specified, the selected unit is used instead.
+                
+
+``-help``:      Shows this help information
+
+
+]====]
+
+local repeatUtil = require('repeat-util')
+local utils = require('utils')
+local validArgs = utils.invert({
+    'add',
+    'del',
+    'clear',
+    'unit',
+    'list',
+    'help',
+})
+local args = utils.processArgs({...}, validArgs)
+
+
+local id = "superdwarf"
+
+superdwarfIds = {}
```

Comment:
```suggestion
superdwarfIds = superdwarfIds or {}
```

This change ensures that the table is not overwritten every time the script is invoked, which looks like it is necessary for `DeleteSuperdwarf()` and `ListSuperdwarfs()` (among others) to work at all.

Reply:
Realized this while testing. Not super familiar with ruby, but realized the `||=` used in the initialization is in fact `var = var || value`

Reply:
yep! unfortunately Lua does not have compound-assignment operators, so anything like `a += b` will always have to translate to `a = a + b`